src/navbar/search-input.tsx

```diff
@@ -0,0 +1,311 @@
+import React, { useEffect, useRef, useState } from "react";
+import { TextWeight } from "../text";
+import { IconButton } from "../icon-button/icon-button";
+
+import {
+    CloseIconButton,
+    CloseIconContainer,
+    Container,
+    CrossIconClose,
+    Divider,
+    DividerVertical,
+    DividerVerticalMobile,
+    DropDownBar,
+    Link,
+    MenuItem,
+    MobileWrapper,
+    SearchBarContainer,
+    SearchBarDesktop,
+    SearchBarInputContainer,
+    SearchIcon,
+    SearchIconButton,
+    SearchInputComponent,
+    SearchInputContainer,
+    SearchMainBarContainer,
+    SearchMobileIcon,
+    SearchSpan,
+} from "./search-input.styles";
+import { NavItemCommonProps } from "./types";
+
+interface Props<T> {
+    items: NavItemCommonProps<T>[];
+    /** toggle for mobile or desktop view */
+    mobile?: boolean | undefined;
+    onItemClick: (
+        event: React.MouseEvent<HTMLAnchorElement>,
+        item: NavItemCommonProps<T>
+    ) => void;
+    onToggleHeaderClick: () => void;
+    onSearchButtonClick?: (searchInput: string) => void;
+}
+
+export const SearchInput = <T,>({
+    items,
+    mobile = false,
+    onItemClick,
+    onToggleHeaderClick,
+    onSearchButtonClick,
+}: Props<T>): JSX.Element => {
+    // =============================================================================
+    // STATE DECLARATIONS
+    // =============================================================================
+    const [toggleInput, setToggleInput] = useState<boolean>(false);
+    const [inputValue, setInputValue] = useState<string>("");
+    const [itemsLocal, setItemsLocalValue] = useState<NavItemCommonProps<T>[]>(
+        []
+    );
+    const [toggleDropdown, setToggleDropdown] = useState<boolean>(false);
+    const [showDropdown, setShowDropdown] = useState<string>("on-blur");
```

Comment:
as discussed
- can simplify this to a `focused` (meaning either blue outline or default grey) and `expanded` state
- can try to reuse `InputSelectWrapper`. for the case where input is greyed out when there are no search results, you can override the default `:focus-within` style